### ISSUE TITLE
min and maxTerrainAltitude seem to still operate based on altitude ASL

### ISSUE THREAD

**Issue by space-is-hard** [jrossignol/ContractConfigurator#335]
Trying to implement a kOS contract pack, the prototype contract is a simple hover that should encourage developing a PID controller. However, we'd like it based off of altitude AGL so that the player can't simply park a rover on a hillside at 500m ASL to complete the contract. Unfortunately, specifying `max/minTerrainAltitude` parameters still behaves as though you're only using `max/minAltitude` parameters.

The contract, for reference:

```
CONTRACT_TYPE
{
    name = kOS_hover_easy
    
    //Contract text
    title = Autonomous hover at a specific altitude
    description = Design a vehicle that will autonomously hover at a specific altitude for a specific period of time
    synopsis = We'd like you to design a vehicle that can hover at a specific altitude above ground level for a specific period of time under the autonomous control of a kOS script.
    completedMessage = Congratulations! You've successfully hovered at the desired altitude for the required amount of time!
    
    //Agency -- Do we make our own or let CC assign a random one?
    //agent = 
    
    //Expiry
    minExpiry = 500.0
    maxExpiry = 1000.0
    
    //Deadline
    deadline = 500.0
    
    //Declinability/Cancellability
    cancellable = true
    declinable = true
    
    //Prestige -- Defaults to any
    prestige = Trivial
    
    //Target celestial body -- defaults to null/none
    //targetBody = Kerbin
    
    //Maximum times contract can be completed -- default is 0/unlimited
    maxCompletions = 0
    
    //Maximum simultaneous instances of this contract -- default is 0/unlimited
    maxSimultaneous = 1
    
    //Rewards
    rewardScience = 20.0
    rewardReputation = 10.0
    rewardFunds = 25000.0
    failureReputation = 10.0
    failureFunds = 10000.0
    advanceFunds = 10000.0
    
    //Completion parameters
    PARAMETER
    {
        name = altitude_parameter
        type = ReachState
        targetBody = Kerbin
        //Craft must hover at 500 meters AGL (+/- 10 meters)
        minTerrainAltitude = 490.0
        maxTerrainAltitude = 510.0
        //Craft must be flying at no more than 10 m/s surface velocity
        maxSpeed = 10.0
        situation = FLYING
        title = Altitude
        
        PARAMETER
        {
            name = time_parameter
            type = Duration
            //Craft must hover in the altitude range for 30s
            duration = 30s
            preWaitText = Waiting for altitude range
            waitingText = Hover timer running
            completionText = Time requirement reached
        }
    }
    
    //This fails for some reason
    // PARAMETER
    // {
        // name = no_crew_parameter
        // type = HasCrewCapacity
        //Craft must not be able to hold crew
        // maxCapacity = 0
    // }
    
    //Contract availibility requirements
    REQUIREMENT
    {
        name = core_part_requirement
        type = PartModuleUnlocked
        //A part with the kOSProcessor module must be unlocked
        partModule = kOSProcessor
    }
    
    REQUIREMENT
    {
        name = liquid_engine_requirement
        type = Any
        //An LV-T45 or LV-T30 liquid engine must be unlocked
        
        REQUIREMENT
        {
            name = LVT45_requirement
            type = PartUnlocked
            part = liquidEngine2  //**VERIFY THAT THIS IS CORRECT**
        }
        
        REQUIREMENT
        {
            name = LVT30_requirement
            type = PartUnlocked
            part = liquidEngine  //**VERIFY THAT THIS IS CORRECT**
        }
    }
    
    REQUIREMENT
    {
        name = probe_core_requirement
        type = PartUnlocked
        //A probe core must be unlocked
        part = probeCoreSphere
    }
    
    REQUIREMENT
    {
        name = altitude_requirement
        type = AltitudeRecord
        //The player must have breached 30km
        minAltitude = 30000
    }
}
```

**Comment by jrossignol** [comment]
Hmmm...  I guess not too many people are using the altitude above ground for this one to sneak through.  I'll take a look at this and get it fixed for the next release.

**Comment by space-is-hard** [comment]
Thanks

**Comment by jrossignol** [comment]
Looks like this was never working correctly since it was added (0.7.0).

I won't be able to test this until tonight, but the change seemed trivial.  Let me know if you're able to verify against the dll [here](https://github.com/jrossignol/ContractConfigurator/blob/db7ab86b66aed68a30e3ff323715adbf1b11e974/GameData/ContractConfigurator/ContractConfigurator.dll?raw=true) before then.

**Comment by space-is-hard** [comment]
No bueno, won't recognize either AGL _or_ ASL anymore

**Comment by jrossignol** [comment]
ASL shouldn't have been touched at all - when you say it's not working, you
mean the parameter isn't getting checked off?

**Comment by space-is-hard** [comment]
Yes, with the new .dll, when the contract asks for a specific terrainAltitude, the parameter isn't checked off when the ship meets that altitude above neither sea level, nor ground level.

**Comment by jrossignol** [comment]
Should be working now ([dll link](https://github.com/jrossignol/ContractConfigurator/blob/78f8f63dd600bbb16ff1f918e198759d5a74583f/GameData/ContractConfigurator/ContractConfigurator.dll?raw=true)).  I'd accidentally used the field that means "terrain's height above sea level" instead of "vessel's height above terrain". :wink:

**Comment by space-is-hard** [comment]
That one works correctly.

**Comment by jrossignol** [comment]
Great, thanks for verifying it.

Fixed for 1.7.8 release.
